validators/ecdsa.go

```diff
@@ -0,0 +1,230 @@
+package validators
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+type ECDSAValidator struct {
+	Address types.Address
+}
+
+func NewECDSAValidator(addr types.Address) *ECDSAValidator {
+	return &ECDSAValidator{
+		Address: addr,
+	}
+}
+
+func ParseECDSAValidator(s string) (*ECDSAValidator, error) {
+	bytes, err := hex.DecodeString(strings.TrimPrefix(s, "0x"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &ECDSAValidator{
+		Address: types.BytesToAddress(bytes),
+	}, nil
+}
+
+func (v *ECDSAValidator) Type() ValidatorType {
+	return ECDSAValidatorType
+}
+
+func (v *ECDSAValidator) String() string {
+	return v.Address.String()
+}
+
+func (v *ECDSAValidator) Addr() types.Address {
+	return v.Address
+}
+
+func (v *ECDSAValidator) Copy() Validator {
+	return &ECDSAValidator{
+		Address: v.Address,
+	}
+}
+
+func (v *ECDSAValidator) Equal(vr Validator) bool {
+	vv, ok := vr.(*ECDSAValidator)
+	if !ok {
+		return false
+	}
+
+	return v.Address == vv.Address
+}
+
+func (v *ECDSAValidator) MarshalRLPWith(arena *fastrlp.Arena) *fastrlp.Value {
+	vv := arena.NewArray()
+
+	vv.Set(arena.NewBytes(v.Address[:]))
+
+	return vv
+}
+
+func (v *ECDSAValidator) UnmarshalRLPFrom(p *fastrlp.Parser, val *fastrlp.Value) error {
+	elems, err := val.GetElems()
+	if err != nil {
+		return err
+	}
+
+	if len(elems) < 1 {
+		return fmt.Errorf("incorrect number of elements to decode ECDSAValidator, expected 1 but found %d", len(elems))
+	}
+
+	if err := elems[0].GetAddr(v.Address[:]); err != nil {
+		return fmt.Errorf("failed to decode Address: %w", err)
+	}
+
+	return nil
+}
+
+func (v *ECDSAValidator) Bytes() []byte {
+	return types.MarshalRLPTo(v.MarshalRLPWith, nil)
+}
+
+func (v *ECDSAValidator) SetFromBytes(input []byte) error {
+	return types.UnmarshalRlp(v.UnmarshalRLPFrom, input)
+}
```

Comment:
Why do you need the RLP format?

Reply:
Validators is put in IBFT Extra and Validator may be set in Header.

Reply:
Jesus Christ